src/Fixer/FunctionNotation/CombineDirnameFixer.php

```diff
@@ -0,0 +1,231 @@
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Replace multiple nested calls of dirname() by only one call with second level parameter. Requires PHP >= 7.0.',
+            [
+                new VersionSpecificCodeSample(
+                    "<?php\ndirname(dirname(dirname(\$path)));\n",
+                    new VersionSpecification(70000)
+                ),
+            ],
+            null,
+            'Risky when dirname() is overridden.'
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return \PHP_VERSION_ID >= 70000 && $tokens->isTokenKindFound(T_STRING);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isRisky()
+    {
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        // should run after DirConstantFixer
+        // should run before MethodArgumentSpaceFixer, NoSpacesInsideParenthesisFixer, NoTrailingWhitespaceFixer, NoWhitespaceInBlankLineFixer
+        return 3;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        for ($index = $tokens->count() - 1; 0 <= $index; --$index) {
+            $token = $tokens[$index];
+
+            if (!$token->equals([T_STRING, 'dirname'], false)) {
+                continue;
+            }
+
+            $dirnameInfo = $this->getDirnameInfo($tokens, $index);
+
+            if (!$dirnameInfo) {
+                continue;
+            }
+
+            $prev = $tokens->getPrevMeaningfulToken($dirnameInfo['indexes'][0]);
+
+            if (!$tokens[$prev]->equals('(')) {
+                continue;
+            }
+
+            $prev = $tokens->getPrevMeaningfulToken($prev);
+
+            $firstArgumentEnd = $dirnameInfo['end'];
+
+            $dirnameInfos = [$dirnameInfo];
+
+            while ($dirnameInfo = $this->getDirnameInfo($tokens, $prev, $firstArgumentEnd)) {
+                $dirnameInfos[] = $dirnameInfo;
+
+                $prev = $tokens->getPrevMeaningfulToken($dirnameInfo['indexes'][0]);
+
+                if (!$tokens[$prev]->equals('(')) {
+                    break;
+                }
+
+                $prev = $tokens->getPrevMeaningfulToken($prev);
+                $firstArgumentEnd = $dirnameInfo['end'];
+            }
+
+            if (count($dirnameInfos) > 1) {
+                $this->combineDirnames($tokens, $dirnameInfos);
+            }
+
+            $index = $prev;
+        }
+    }
+
+    /**
+     * @param Tokens   $tokens
+     * @param int      $index                 Index of `dirname`
+     * @param null|int $firstArgumentEndIndex Index of last token of first argument of `dirname` call
+     *
+     * @return array|bool `false` when it is not a (suppoerted) `dirname` call, an array with infos about the dirname call otherwise
```

Comment:
Typo: *suppo**e**rted* => *supported*. Also *info* (*information*) is uncountable in english and cannot have an *s*.

Reply:
Thanks, fixed it.